docs/src/contribute/design/next-tedge-api/registration.md

```diff
@@ -0,0 +1,227 @@
+---
+title: Registration
+tags: [Documentation]
+sidebar_position: 3
+---
+
+## Overview
+
+Automatic registration relies on user-defined configuration to provide a link between the topic names and the entity/component types.
+
+The only difference between automatic and manual registration, a thin-edge component takes over the publishing of the registration message for the other entity/component rather than relying on a manual registration to have already occurred.
+
+Once an entity/component has been automatically registered, then it will behave just like a manually registered entity/component.
+
```

Comment:
In the case of `manual` registration the `device id` can be provided by the user, but in the case of `automatic` registration the `device-id` is derived from the topic fragments right?

Reply:
Yes, though the component doing the auto registration logic (e.g. deriving the device id from the topic name), only should do this one and then create a registration event which in turn would add the topic-to-entity mapping to the entity store (so it does not always have to keep on deriving the device id each time). This keeps the work of the mapper very consistent regardless if there is an entity is registered via automatic or manual methods.